Clarify conditions under which process.env expressions are inlined (#33212)
vercel/next.js: docs/basic-features/environment-variables.md

NEXT_PUBLIC_ANALYTICS_ID=abcdefghijk
```

> This loads `process.env.NEXT_PUBLIC_ANALYTICS_ID` into the Node.js environment automatically, allowing you to use it anywhere in your code. To expose these variables to JavaScript on the browser, any JavaScript expressions of the form `process.env.NEXT_PUBLIC_FOO` will be inlined at build time, replaced with the string value of the `NEXT_PUBLIC_FOO` environment variable set when the project is built. For example:

Reviewer: I don't think this reads any better. In fact, it assumes the reader understands what `FOO` means. Let's revert this paragraph and keep the comment on line 98.

Reviewer 2: Linguistics are easy to bike shed on - I think the original is maybe slightly better linguistically. But I think the _key_ thing that @jameshfisher is trying to emphasize here is that the build process will look for `process.env.NEXT_PUBLIC_FOO`. We could also write `process.env.NEXT_PUBLIC_*` or something like that.

> This loads `process.env.NEXT_PUBLIC_ANALYTICS_ID` into the Node.js environment automatically, allowing you to use it anywhere in your code. To expose these variables to JavaScript on the browser, any JavaScript expressions of the form `process.env.NEXT_PUBLIC_FOO` will be inlined at build time, replaced with the string value of the `NEXT_PUBLIC_FOO` environment variable set when the project is built. For example:

Reviewer 3: ```suggestion
This loads `process.env.NEXT_PUBLIC_ANALYTICS_ID` into the Node.js environment automatically, allowing you to use it anywhere in your code. The value will be inlined into JavaScript sent to the browser because of the `NEXT_PUBLIC_` prefix. This inlining occurs at build time, so your various `NEXT_PUBLIC_` envs need to be set when the project is built.
```

export default HomePage
```

Note that dynamic lookups will _not_ be inlined, such as:

```js

> // This will not be inlined, because it uses a variable

Reviewer: ```suggestion
// This will NOT be inlined, because it uses a variable
```

const varName = 'NEXT_PUBLIC_ANALYTICS_ID'
setupAnalyticsService(process.env[varName])

> // This will not be inlined, because it uses a variable

Reviewer: ```suggestion
// This will NOT be inlined, because it uses a variable
```
